clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -361,69 +172,35 @@ private static boolean shouldRetainsBufferReference(Schema[] requestSchemas) {
     }
 
     public static ApiKeys forId(int id) {
```

Comment:
Is 'short' type more suitable?

Reply:
If the above comment is valid, it can be addressed in separate PR :)

Reply:
We used `int` here because `short` in Java is a second class citizen and casting is required in some cases. If we decide to change this, a separate PR (as you suggested) would be better.